api/api.go

```diff
@@ -27,7 +29,7 @@ type (
 )
 
 // New returns a new initialised API.
-func New(db *database.DB, logger *logrus.Logger) (*API, error) {
+func New(cfg conf.Config, db *database.DB, logger *logrus.Logger) (*API, error) {
```

Comment:
we don't generally inject high level config structs that contain all configuration into services but instead only supply the service with the information it needs and nothing more

Reply:
I don't really mind it's just that doing it this way it becomes harder to see what services require what configuration values.

Reply:
that's reasonable. changed to pass only what a component needs.